File: const/stations.js
const Station = require("./Station.js");
const XLS = require("xlsx");
const path = require('path');

const curr = path.dirname(__filename);
const filePath = path.join(curr, 'stations.xls');
const workbook = XLS.readFile(filePath);

const sheetName = workbook.SheetNames[0];
const worksheet = workbook.Sheets[sheetName];

const data = XLS.utils.sheet_to_json(worksheet);

const getStations = () => {
  const stations = {};
  data.forEach((stnObj) => {
    const station = new Station(
      stnObj["stn_code"].trim(),
      stnObj["mrt_station_english"].trim(),
      stnObj["mrt_station_chinese"].trim(),
      stnObj["mrt_line_english"].trim(),
      stnObj["mrt_line_chinese"].trim()
    );
    const key = station.getLine();

    if (!stations[key]) {
      stations[key] = [station];
    } else {
      stations[key].push(station);
    }
  });

  return stations;
};

const getStnEdges = () => {
  const stations = getStations();
  const stnList = [];

  for (const key in stations) {
    const line = stations[key];
    for (let i = 0; i < line.length - 1; i++) {
      const stn = line[i].name;
      const nextStn = line[i + 1].name;
      stnList.push([stn, nextStn]);
    }
  }
  console.log(stnList.length);

  return stnList;
};

module.exports = getStnEdges;

File: const/stations.py
import pandas as pd
from .Station import Station
import os

current_dir = os.path.dirname(os.path.abspath(__file__))
file_path = os.path.join(current_dir, 'stations.xls')
# file_path = 'stations.xls'
df = pd.read_excel(file_path)
df = df.apply(lambda x: x.str.strip() if x.dtype == "object" else x)

def getStations():
    stations = {}
    for index, row in df.iterrows():
        station = Station(row['stn_code'], row['mrt_station_english'], row['mrt_station_chinese'], row['mrt_line_english'], row['mrt_line_chinese'])
        line, chiLine = station.getLine(), station.getChiLine()
        key = (line, chiLine)
        if key not in stations:
            stations[key] = [station]
        else:
            stations[key].append(station)
    # print(stations[("North-South Line", "南北线")])
    return stations

# def getStnEdges():
#     stn_dict = getStations()
#     stn_list = []
    
#     for key in stn_dict:
#         # print(key, len(stn_dict[key]))
#         line = stn_dict[key]
#         for i in range(len(line)-1):
#             stnObj = line[i]
#             nextStnObj = line[i+1]
#             stn_list.append([stnObj, nextStnObj])

#     return stn_list

def getStnEdges():
    stn_dict = getStations()
    stn_list = []
    
    for key in stn_dict:
        line = stn_dict[key]
        for i in range(len(line)-1):
            stn = line[i].getName()
            nextStn = line[i+1].getName()
            stn_list.append([stn, nextStn])

    return stn_list